pyro/poutine/trace_struct.py

```diff
@@ -148,14 +148,18 @@ def compute_log_prob(self, site_filter=lambda name, site: True):
                 try:
                     site["log_prob"]
                 except KeyError:
-                    log_p = site["fn"].log_prob(site["value"], *site["args"], **site["kwargs"])
-                    site["unscaled_log_prob"] = log_p
-                    log_p = scale_and_mask(log_p, site["scale"], site["mask"])
-                    site["log_prob"] = log_p
-                    site["log_prob_sum"] = log_p.sum()
-                    if is_validation_enabled():
-                        warn_if_nan(site["log_prob_sum"], "log_prob_sum at site '{}'".format(name))
-                        warn_if_inf(site["log_prob_sum"], "log_prob_sum at site '{}'".format(name), allow_neginf=True)
+                    try:
+                        log_p = site["fn"].log_prob(site["value"], *site["args"], **site["kwargs"])
+                        site["unscaled_log_prob"] = log_p
+                        log_p = scale_and_mask(log_p, site["scale"], site["mask"])
+                        site["log_prob"] = log_p
+                        site["log_prob_sum"] = log_p.sum()
+                        if is_validation_enabled():
+                            warn_if_nan(site["log_prob_sum"], "log_prob_sum at site '{}'".format(name))
+                            warn_if_inf(site["log_prob_sum"], "log_prob_sum at site '{}'".format(name),
+                                        allow_neginf=True)
+                    except ValueError as e:
+                        raise ValueError("Error while computing log_prob at site '{}'".format(name)) from e
```

Comment:
Unfortunately `raise ... from ...` is Python 3 only: https://stackoverflow.com/questions/27318327/how-to-imitate-python-3s-raise-from-in-python-2

Reply:
Nice, you should be able to use `six.raise_from` in Pyro, as discussed in Eli's comment

Reply:
It seems that `six.raise_from` delegates to `raise exc` for python-2.7, but [six.reraise](https://pythonhosted.org/six/#six.reraise)(*sys.exc_info()) might help with this use case. Could you test your example on python 2 as well?

Reply:
Thanks for all the suggestions! I ran the tests locally and then somehow missed that the code should be Python 2 compatible as well.

I will make sure to make it backwards compatible and test it with Python 2 ASAP.